client/landing/jetpack-cloud/components/date-picker/index.jsx

```diff
@@ -18,80 +19,100 @@ import Gridicon from 'components/gridicon';
 
-const DATE_FORMAT = 'YYYY-MM-DD';
-
 class DatePicker extends Component {
 	static propTypes = {
 		siteId: PropTypes.number.isRequired,
-		selectedDateString: PropTypes.string.isRequired,
-		onChange: PropTypes.func.isRequired,
+		selectedDate: PropTypes.instanceOf( Date ).isRequired,
+		onDateChange: PropTypes.func.isRequired,
+		onDateRangeSelection: PropTypes.func.isRequired,
 	};
 
-	getFormattedDate = dateString => this.props.moment.parseZone( dateString ).format( DATE_FORMAT );
+	getDisplayDate = ( date, showTodayYesterday = true ) => {
+		const { moment, translate } = this.props;
+
+		const daysDiff = moment().diff( date, 'days' );
+		const yearToday = moment().format( 'YYYY' );
+		const yearDate = moment( date ).format( 'YYYY' );
+
+		let dateFormat = 'MMM D';
+
+		if ( yearToday !== yearDate ) {
+			dateFormat = 'MMM D, YYYY';
+		}
 
-	getDisplayDate = dateString => {
-		const word = this.props.moment
-			.parseZone( dateString )
-			.calendar()
-			.split( ' ' )[ 0 ];
-		if ( 'Today' === word || 'Yesterday' === word ) {
-			return word;
+		if ( showTodayYesterday ) {
+			switch ( daysDiff ) {
+				case 0:
+					return translate( 'Today' );
+				case 1:
+					return translate( 'Yesterday' );
+			}
 		}
 
-		return this.getFormattedDate( dateString );
+		return moment( date ).format( dateFormat );
 	};
 
 	shuttleLeft = () => {
-		const { moment, onChange, selectedDateString } = this.props;
-		const newDateString = moment
-			.parseZone( selectedDateString )
-			.subtract( 1, 'days' )
-			.toISOString( true );
+		const { moment, onDateChange, selectedDate } = this.props;
 
-		onChange( newDateString );
+		const newSelectedDate = moment( selectedDate ).subtract( 1, 'days' );
+
+		onDateChange( newSelectedDate.toDate() );
 	};
 
 	shuttleRight = () => {
 		if ( ! this.canShuttleRight() ) {
 			return false;
 		}
 
-		const { moment, onChange, selectedDateString } = this.props;
-		const newDateString = moment
-			.parseZone( selectedDateString )
-			.add( 1, 'days' )
-			.toISOString( true );
+		const { moment, onDateChange, selectedDate } = this.props;
+
+		const newSelectedDate = moment( selectedDate ).add( 1, 'days' );
 
-		onChange( newDateString );
+		onDateChange( newSelectedDate.toDate() );
 	};
 
 	canShuttleRight = () => {
-		const { moment, selectedDateString } = this.props;
-		return ! moment().isSame( moment.parseZone( selectedDateString ), 'day' );
+		const { moment, selectedDate } = this.props;
+
+		return ! moment( selectedDate ).isSame( moment(), 'day' );
 	};
 
 	render() {
-		const { selectedDateString, siteId } = this.props;
+		const { selectedDate, siteId, moment } = this.props;
+
+		const previousDate = moment( selectedDate ).subtract( 1, 'days' );
+		const nextDate = moment( selectedDate ).add( 1, 'days' );
 
-		const currentDisplayDate = this.getDisplayDate( selectedDateString );
+		const previousDisplayDate = this.getDisplayDate( previousDate );
+		const nextDisplayDate = this.getDisplayDate( nextDate, false );
```

Comment:
<img width="217" alt="Screen Shot 2020-03-16 at 10 35 56 AM" src="https://user-images.githubusercontent.com/2810519/76785028-ec33bb00-6771-11ea-8223-d58bb4a0300d.png">
I feel like in this case "Mar 16" should say "today"

Reply:
No, I commented it with Yvonne, and she told me that in this case, it is weird to see a "future date" with the word Today.